sklearn/utils/estimator_checks.py

```diff
@@ -1748,7 +1775,9 @@ def check_classifier_multioutput(name, estimator, strict_mode=True):
 
 
 @ignore_warnings(category=FutureWarning)
-def check_regressor_multioutput(name, estimator, strict_mode=True):
+def check_regressor_multioutput(name, estimator, api_only=False):
```

Comment:
we can make the shape an API check and leave the dtype out of the API check? 
This test also has `not api_check` sections.